blockchain/validator_set.go

```diff
@@ -0,0 +1,273 @@
+package blockchain
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"math"
+	"sort"
+	"strconv"
+
+	acm "github.com/hyperledger/burrow/account"
+	rpc_core "github.com/tendermint/tendermint/rpc/core"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tm_types "github.com/tendermint/tendermint/types"
+)
+
+type ValidatorSet interface {
+	TotalPower() int
+	SetMaximumPower(maximumPower int)
+	MaximumPower() int
+	AdjustPower(height int64) error
+	Validators() []acm.Validator
+	SetLeavers() []acm.Validator
+	JoinToTheSet(validator acm.Validator) error
+	LeaveFromTheSet(validator acm.Validator) error
+	IsValidatorInSet(address acm.Address) bool
+	JSONBytes() ([]byte, error)
+}
+
+type validatorListProxy interface {
+	Validators(height int64) (*ctypes.ResultValidators, error)
+}
+
+type _validatorListProxy struct{}
+
+func (vlp _validatorListProxy) Validators(height int64) (*ctypes.ResultValidators, error) {
+	result, err := rpc_core.Validators(&height)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+type validatorSet struct {
+	maximumPower int
+	validators   []acm.Validator
+	setLeavers   []acm.Validator
+	proxy        validatorListProxy
+}
+
+func newValidatorSet(maximumPower int, validators []acm.Validator) *validatorSet {
+	set := &validatorSet{
+		validators: validators,
+		proxy:      _validatorListProxy{},
+	}
+	set.SetMaximumPower(maximumPower)
+	return set
+}
+
+func (vs *validatorSet) TotalPower() int {
+	return len(vs.validators)
+}
+
+func (vs *validatorSet) SetMaximumPower(maximumPower int) {
+	if maximumPower > 90 {
+		maximumPower = 90
```

Comment:
This looks like a stub, but would prefer constants for the magic numbers at least

Reply:
Thanks.

Reply:
That's would be good since it reflects your 1 unit of power for each validator design.